Source/Core/Runtime/SceneObjects/ProcessSceneObject.cs

```diff
@@ -82,44 +83,72 @@ protected void Awake()
             }
         }
 
+        public void SetUniqueId(Guid guid)
+        {
+            uniqueId = guid.ToString();
+        }
+
         private void Reset()
         {
             Init();
         }
 
-        protected void Init()
-        {
 #if UNITY_EDITOR
-            if (UnityEditor.SceneManagement.EditorSceneManager.IsPreviewScene(gameObject.scene))
-            {
-                Debug.LogError($"Not registering {gameObject.name} due to preview scene.");
-                return;
-            }
-#endif
-            if (RuntimeConfigurator.Exists == false)
+        /// <summary>
+        /// Overriding the Reset context menu entry in order to unregister the object before invalidating the unique id.
+        /// </summary>
+        [ContextMenu("Reset", false, 0)]
+        protected void ResetContextMenu()
+        {
+            if (RuntimeConfigurator.Exists)
             {
-                Debug.LogError($"Not registering {gameObject.name} due to runtime configurator not present.");
-                return;
+                RuntimeConfigurator.Configuration.SceneObjectRegistry.Unregister(this);
             }
 
-            if (IsDuplicateUniqueTag())
+            uniqueId = null;
+            tags = new List<string>();
+            Init();
+
+            UnityEditor.EditorUtility.SetDirty(this);
+        }
+
+        [ContextMenu("Make Unique")]
```

Comment:
"Make Unique" can imply that it will not regenerate it when it is already unique.
```suggestion
        [ContextMenu("Reset Unique ID")]
```

Reply:
For [ContextMenu("Reset", false, 0)] we should either also add a Dialog or what I would prefer retain the original unique id.

Reply:
I would not because Reset is supposed to mimic Unity's default reset function, so it should work the same.